tests/integration/test_utils/test_manager.rs

```diff
@@ -0,0 +1,129 @@
+use std::fmt::Display;
+
+use mockito::Mock;
+use thiserror::Error;
+use trycmd::TestCases;
+
+use crate::integration::{env, MockEndpointBuilder, VERSION};
+
+use super::{mock_common_endpoints, ChunkOptions, ServerBehavior};
+
+#[derive(Error, Debug)]
+pub enum Error {
+    #[error("Failed to insert server variable")]
+    InsertServerVar(#[from] trycmd::Error),
+}
+
+pub type Result<T> = std::result::Result<T, Error>;
+
+/// Helper struct for managing integration tests.
+/// Allows for mocking endpoints and registering different types of tests.
+pub struct TestManager {
+    mocks: Vec<Mock>,
+}
+
+impl TestManager {
+    /// Create a new `TestManager`.
+    /// The test manager has no mocked endpoints by default.
+    pub fn new() -> Self {
+        Self { mocks: vec![] }
+    }
+
+    /// Create a mock endpoint on the mockito test server with the given options.
+    /// Returns the updated `TestManager` with the new mock endpoint.
+    pub fn mock_endpoint(mut self, opts: MockEndpointBuilder) -> Self {
+        self.mocks.push(opts.create());
+        self
+    }
+
+    /// Mock the common upload endpoints.
+    pub fn mock_common_upload_endpoints(
+        self,
+        behavior: ServerBehavior,
+        chunk_options: ChunkOptions,
+    ) -> Self {
+        mock_common_endpoints::common_upload_endpoints(behavior, chunk_options)
+            .fold(self, |manager, builder| manager.mock_endpoint(builder))
+    }
+
+    /// Assert that all mocks have been called the correct number of times.
+    ///
+    /// This method consumes the `TestManager`, thereby also stopping the mocks from being
+    /// served, since after asserting the mocks, they should not be needed anymore.
+    pub fn assert_mock_endpoints(self) {
+        for mock in self.mocks {
+            mock.assert();
+        }
+    }
+
+    /// Register a trycmd test.
+    /// The test is run when the returned `TrycmdTestManager` is dropped.
+    /// Further configuration can be done with the `TrycmdTestManager`.
+    pub fn register_trycmd_test(self, path: impl Display) -> TrycmdTestManager {
+        TrycmdTestManager::new(self, path)
+    }
+}
+
+/// Helper struct for managing trycmd tests.
+/// The tests are run when the `TrycmdTestManager` is dropped.
+pub struct TrycmdTestManager {
+    test_case: TestCases,
+    // Manager must be kept alive to keep the mock server alive. Otherwise, the test
+    // case will fail. Trycmd tests run when the TestCases are dropped. So, order
+    // of the fields is important here.
```

Comment:
@loewenheim and I discussed this offline. I pointed out that the [Rust language specifies](https://doc.rust-lang.org/reference/destructors.html#destructors.operation) that struct fields are dropped in the order of declaration, and that implementing `Drop` is not super-straightforward because `assert_mock_endpoints` would fail to compile after such a change (since we cannot move any fields out of a struct that implements `Drop`).